src/core/utilities/iterators.hpp

```diff
@@ -0,0 +1,89 @@
+
+#ifndef PHARE_CORE_UTILITIES_ITERATORS_H
+#define PHARE_CORE_UTILITIES_ITERATORS_H
+
+#include <type_traits>
+
+namespace PHARE::core
+{
+template<typename T, typename Vector>
+struct wrapped_iterator
```

Comment:
this version, or https://github.com/PhilipDeegan/PHARE/blob/soa/src/core/utilities/iterators.hpp#L13

Reply:
actually I'm considering removing it from this PR as it is not used anymore since particle mapping is via indexing